src/Symfony/Component/Security/Csrf/TokenStorage/CookieTokenStorageListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Csrf\TokenStorage;
+
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+/**
+ * Checks the request's attributes for a CookieTokenStorage instance. If one is
+ * found, the cookies representing the storage's changeset are appended to the
+ * response headers.
+ *
+ * TODO where to put this class?
+ *
+ * @author Oliver Hoff <oliver@hofff.com>
+ */
+class CookieTokenStorageListener
+{
+    /**
+     * @var string
+     */
+    private $tokenStorageKey;
+
+    /**
+     * @param string|null $tokenStorageKey
+     */
+    public function __construct($tokenStorageKey = null)
+    {
+        // TODO should this class get its own DEFAULT_TOKEN_STORAGE_KEY constant?
+        // if no, where should the sole constant be put?
+        $this->tokenStorageKey = $tokenStorageKey === null ? RequestStackTokenStorage::DEFAULT_TOKEN_STORAGE_KEY : $tokenStorageKey;
```

Comment:
I don't understand, I mean that any class implementing `TokenStorageInterface` might need a default storage key so it could make sense to hold this constant in the interface.